src/components/cartdetail/PageCartDetail.vue

```diff
@@ -50,24 +55,66 @@ import gql from 'graphql-tag';
 import cartMixin from '@/mixins/cartMixin';
 import CartContent from '@/components/cartdetail/CartContent.vue';
 import PriceCalculation from '@/components/common/PriceCalculation.vue';
+import DiscountCode from '@/components/common/DiscountCode.vue';
 import DisplayableMoneyFragment from '@/components/DisplayableMoney.gql';
 
 export default {
-  components: { CartContent, PriceCalculation },
+  components: {
+    CartContent,
+    PriceCalculation,
+    DiscountCode,
+  },
 
   data: () => ({
     me: null,
+    buttonState: null,
+    serverError: null,
   }),
 
+  methods: {
+    addDiscountCode(code) {
+      return this.updateMyCart({
+        addDiscountCode: {
+          code,
+        },
+      });
+    },
+
+    removeCode(id) {
```

Comment:
I would call it `removeDiscountCode`, similar to how you named `addDiscountCode`, following the name of the action. The reason is that when I read `removeCode(id)` I may not be sure if you mean discount codes or any other kind of code that might be implemented in this component (as the component's name is not referring to discount codes, i.e. `PageCartDetail`).